HPXMLtoOpenStudio/resources/hotwater_appliances.rb

```diff
@@ -964,6 +965,89 @@ def self.get_fixtures_gpd(eri_version, nbeds, fixtures_all_low_flow, daily_mw_fr
     return f_eff * ref_f_gpd * fixtures_usage_multiplier
   end
 
+  def self.add_showers_and_calculate_max(model, runner, hpxml, weather, water_heating_system, eri_version, schedules_file, unavailable_periods)
+    nbeds = hpxml.building_construction.additional_properties.adjusted_number_of_bedrooms
+
+    # Process fixtures
+    fixtures_usage_multiplier = hpxml.water_heating.water_fixtures_usage_multiplier
+    fixtures_all_low_flow = true
+    hpxml.water_fixtures.each do |water_fixture|
+      next unless [HPXML::WaterFixtureTypeShowerhead, HPXML::WaterFixtureTypeFaucet].include? water_fixture.water_fixture_type
+
+      fixtures_all_low_flow = false if not water_fixture.low_flow
+    end
+
+    # Get distribution system
+    if hpxml.water_heating_systems.size > 0
+      hot_water_distribution = hpxml.hot_water_distributions[0]
+    end
+
+    # Calculate mixed water fractions
+    t_mix = 105.0 # F, Temperature of mixed water at fixtures
+    avg_setpoint_temp = 0.0 # WH Setpoint: Weighted average by fraction DHW load served
+    hpxml.water_heating_systems.each do |water_heating_system|
+      wh_setpoint = water_heating_system.temperature
+      wh_setpoint = Waterheater.get_default_hot_water_temperature(eri_version) if wh_setpoint.nil? # using detailed schedules
+      avg_setpoint_temp += wh_setpoint * water_heating_system.fraction_dhw_load_served
+    end
+    daily_wh_inlet_temperatures = calc_water_heater_daily_inlet_temperatures(weather, nbeds, hot_water_distribution, fixtures_all_low_flow, hpxml.header.sim_calendar_year)
+    daily_mw_fractions = calc_mixed_water_daily_fractions(daily_wh_inlet_temperatures, avg_setpoint_temp, t_mix)
```

Comment:
This is all duplicated code from elsewhere. We should rework this code so that the new calculation & EMS program is simply created in the existing code as we process each water heater (rather than trying to do this all at the end of the measure).

Reply:
yeah, at one point I did have most of this integrated in with the fixtures, but I was having trouble figuring out to structure things. Let me take another shot at that and then if I get stuck I'll ask you or Joe for some help.